reader.py

```diff
@@ -41,4 +41,4 @@ def start_url_crawler(self):
     fp = open("final_urls.txt", "r+")
```

Comment:
Shouldnt there be an encoding while reading too ??

Reply:
yes, encoding is required